src/vs-bicep/Bicep.VSLanguageServerClient/DidChangeWatchedFilesNotifier.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Workspace.VSIntegration.Contracts;
+using StreamJsonRpc;
+
+namespace Bicep.VSLanguageServerClient
+{
+    /// <summary>
+    /// VS lsp doesn't support 'workspace/didChangeWatchedFiles' notification. This class
+    /// watches workspace files with extension - arm, bicep, json, jsonc and sends across
+    /// notification to language server whenever there is a file change/create/delete event.
+    /// </summary>
+    public class DidChangeWatchedFilesNotifier : IDisposable
+    {
+        private readonly string[] filters = { "*.arm", "*.bicep", "*.json", "*.jsonc" };
+        private readonly string? location;
+        private readonly JsonRpc rpc;
+        private List<FileSystemWatcher> fileSystemWatchers = new List<FileSystemWatcher>();
+
+        public DidChangeWatchedFilesNotifier(JsonRpc rpc)
+        {
+            this.rpc = rpc;
+
+            var sVsSolution = ServiceProvider.GlobalProvider.GetService(typeof(SVsSolution).GUID);
+
+            // Handles project system scenario
```

Comment:
Are we handling when a solution/project/folder is closed and another opened?

Reply:
Good question.

AttachForCustomMessageAsync(..) in BicepLanguageServerClient get called on solution/project change events. So we should be good.

Reply:
Included this scenario in gif attached to description.

Reply:
thx